backtester/eventhandlers/statistics/currencystatstics/currencystatistics.go

```diff
@@ -0,0 +1,213 @@
+	gctcommon "github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/math"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	gctkline "github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	gctorder "github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+// CalculateResults calculates all statistics for the exchange, asset, currency pair
+func (c *CurrencyStatistic) CalculateResults() {
+	first := c.Events[0]
+	firstPrice := first.DataEvent.ClosePrice()
+	last := c.Events[len(c.Events)-1]
+	lastPrice := last.DataEvent.ClosePrice()
+	for i := range last.Transactions.Orders {
+		if last.Transactions.Orders[i].Side == gctorder.Buy {
+			c.BuyOrders++
+		} else if last.Transactions.Orders[i].Side == gctorder.Sell {
+			c.SellOrders++
+		}
+	}
+	for i := range c.Events {
+		price := c.Events[i].DataEvent.ClosePrice()
+		if c.LowestClosePrice == 0 || price < c.LowestClosePrice {
+			c.LowestClosePrice = price
+		}
+		if price > c.HighestClosePrice {
+			c.HighestClosePrice = price
+		}
+	}
+	c.MarketMovement = ((lastPrice - firstPrice) / firstPrice) * 100
+	c.StrategyMovement = ((last.Holdings.TotalValue - last.Holdings.InitialFunds) / last.Holdings.InitialFunds) * 100
+	c.RiskFreeRate = last.Holdings.RiskFreeRate * 100
+	returnPerCandle := make([]float64, len(c.Events))
+
+	var negativeReturns []float64
+	var allDataEvents []common.DataEventHandler
+	for i := range c.Events {
+		returnPerCandle[i] = c.Events[i].Holdings.ChangeInTotalValuePercent
+		if c.Events[i].Holdings.ChangeInTotalValuePercent < 0 {
+			negativeReturns = append(negativeReturns, c.Events[i].Holdings.ChangeInTotalValuePercent)
+		}
+		allDataEvents = append(allDataEvents, c.Events[i].DataEvent)
+	}
+
+	c.MaxDrawdown = calculateMaxDrawdown(allDataEvents)
+
+	interval := first.DataEvent.GetInterval()
+	intervalsPerYear := interval.IntervalsPerYear()
+	durationPerYear := intervalsPerYear * float64(interval.Duration())
+	btDuration := float64(last.DataEvent.GetTime().Sub(first.DataEvent.GetTime()))
+
+	relativelyRiskFree := first.Holdings.RiskFreeRate / intervalsPerYear
+
+	c.SharpeRatio = math.CalculateSharpeRatio(returnPerCandle, relativelyRiskFree)
+	c.SortinoRatio = math.CalculateSortinoRatio(returnPerCandle, negativeReturns, relativelyRiskFree)
+	c.InformationRatio = math.CalculateInformationRatio(returnPerCandle, []float64{relativelyRiskFree})
+	c.CalmarRatio = math.CalculateCalmarRatio(returnPerCandle, c.MaxDrawdown.Highest.Price, c.MaxDrawdown.Lowest.Price)
+
+	c.CompoundAnnualGrowthRate = math.CalculateCompoundAnnualGrowthRate(
+		last.Holdings.InitialFunds,
+		last.Holdings.TotalValue,
+		durationPerYear,
+		btDuration)
+}
+
+// PrintResults outputs all calculated statistics to the command line
+func (c *CurrencyStatistic) PrintResults(e string, a asset.Item, p currency.Pair) {
```

Comment:
Request: Just in the README add a small explanation and what kind of values you might expect from each ratio for new people. Does not need to be done for this PR.